spec/controllers/orders_controller_spec.rb

```diff
@@ -734,9 +734,9 @@ def setup_past_reservation
               expect(assigns[:order].order_details).to all(be_canceled)
             end
 
-            it "sets the canceled time on the reservation" do
-              assigns[:order].order_details.all? { |od| expect(od.reservation.canceled_at).not_to be_nil }
-              expect(reservation.reload.canceled_at).not_to be_nil
+            it "sets the canceled time on the order_detail" do
+              assigns[:order].order_details.all? { |od| expect(od.canceled_at).not_to be_nil }
+              expect(reservation.order_detail.reload.canceled_at).not_to be_nil
```

Comment:
Nope, no direct access, at least not for the order_detail that this reservation belongs to.